src/resource/prelude.rs

```diff
@@ -0,0 +1,87 @@
+//! Prelude extension - available in all contexts, defining common types,
+//! operations and constants.
+use lazy_static::lazy_static;
+use smol_str::SmolStr;
+
+use crate::{
+    resource::TypeDefBound,
+    types::{
+        type_param::{TypeArg, TypeParam},
+        CustomType, Type, TypeBound,
+    },
+    Resource,
+};
+
+lazy_static! {
+    /// Prelude resource
+    pub static ref PRELUDE: Resource = {
+        let mut prelude = Resource::new(SmolStr::new_inline("prelude"));
+        prelude
+            .add_type(
+                SmolStr::new_inline("float64"),
+                vec![],
+                "float64".into(),
```

Comment:
Are we removing the `arithmetic.float.types` resource (which also defines a "float64" type)? Or should that (and maybe all of the arithmetic resources) move into the prelude?

Reply:
None of the tests actually use it so I'm just removing it - what exactly the
prelude should contain can wait (for the purposes of this PR it is types
required for existing tests)